### PR TITLE
PHPC-2649: Use output from setup-php-sdk in packaging step

### PR DESCRIPTION
**Pull request by alcaeus**
PHPC-2649

This PR adds the `vs` output variable from the `setup-php-sdk` action to all actions involved in building Windows packages so that it can be used when determining the filename for the archive. This removes the need for maintaining a list of PHP versions with their compilers.

In addition to this, the PR makes two more adjustments for compatibility with PIE:
* The archive name should contain `x86_64` for 64-bit platforms, not `x64`
* The built DLL should be named like the archive instead of the default `php_mongodb.dll`. PIE will rename the file in its install step.

cc @mickverm - this should get rid of the issues you encountered, but we won't know until we release 1.20.1. Please bear with us while we sort out the initial difficulties with PIE. We do appreciate your testing efforts for PIE compatibility 💪

### CODE REVIEW

**Comment by mickverm** [comment]
I'll test it out as soon as a new version is released, thank you for keeping me up to date. 😄

File: .github/workflows/build-windows-package.yml
Comment (mickverm) [inline .github/workflows/build-windows-package.yml:132]: If I'm not mistaken, `php_mongodb.pdb` needs to become `${{ env.FILENAME }}.pdb`.

`php_mongodb-1.20.1-8.3-ts-vs16-x86_64.dll` would be converted to `php_mongodb-1.20.1-8.3-ts-vs16-x86_64.pdb` in https://github.com/php/pie/blob/eaf77ea181f44c7711887474c4f33501a17cb39f/src/Installing/WindowsInstall.php#L39

Comment (alcaeus) [reply .github/workflows/build-windows-package.yml:132]: Hmm, according to the [maintainer docs](https://github.com/php/pie/blob/main/docs/extension-maintainers.md#contents-of-the-windows-zip), it doesn't:

> `php_{extension-name}.pdb` - this will be moved alongside the `C:\path\to\php\ext\php_{extension-name}.dll`

Looking at the code you linked, I do agree that it should be named the same way as the DLL file.

Let me try summoning @asgrim to clarify 😉 

---

Another concern that comes to mind is that we're now requiring people that install their extensions manually. Right now, they only need to copy `php_mongodb.dll` to their extensions directory and they're done. This is valid regardless of whether they download our packages or the ones offered on pecl.php.net.

I'm sure there was a particular reason for naming the DLL file this way, but I wonder if that's truly necessary. Clearly, we can extract the relevant information (compiler, thread safety, PHP version, architecture) from the archive filename before we extract the DLL file, so we shouldn't need to re-verify this when we have the DLL file. If a user packs a DLL file into an archive for a different platform, that's their problem, not PIE's.

Comment (asgrim) [reply .github/workflows/build-windows-package.yml:132]: Regarding the first part, yes the implementation is incorrect, it was always meant to be `php_{extension-name}.pdb` - I've reported issue php/pie#71 and I will look into getting this fixed.

The second part; the naming was agreed since it aligns with what [php/php-windows-builder](https://github.com/php/php-windows-builder) action produces

File: .github/workflows/build-windows-package.yml
Comment (jmikola) [inline .github/workflows/build-windows-package.yml:117]: Noted that this uses the [`needs` context](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#needs-context).

**Review by jmikola — APPROVED** [review]
LGTM assuming `php_mongodb.pdb` is using the correct name.

**Pull request merged**
